lib/cln.js

```diff
@@ -3,8 +3,16 @@ import https from 'https'
 import crypto from 'crypto'
 
 export const createInvoice = async ({ socket, rune, cert, label, description, msats, expiry }) => {
-  const agent = cert ? new https.Agent({ ca: Buffer.from(cert, 'base64') }) : undefined
-  const url = 'https://' + socket + '/v1/invoice'
+  let scheme = 'https:'
+  if (/\.onion(:[0-9]+)?$/.test(socket)) {
+    // we force HTTP because we don't support HTTPS over Tor yet
```

Comment:
We don't support Tor at all with fetch yet right?

Reply:
Oh, right, `node-fetch` doesn't care about `http_proxy`: https://github.com/node-fetch/node-fetch/issues/195

So this wouldn't work as is.

I found [this library](https://github.com/delvedor/hpagent) now [which extends `https.Agent`](https://github.com/delvedor/hpagent/blob/main/index.js#L64).

Reply:
Closed in favor of https://github.com/stackernews/stacker.news/pull/1136